### PR TITLE
Requirements cleanup

### PR DESCRIPTION
**Pull request by scotthavens**
The `requirement.txt` was a bit of a mess with a lot of unneeded packages. Issue #111 outlined the package dependencies that could be removed and the requirements were further split into the base requirements and those needed for testing and the documentation.

Along with this, the `Dockerfile` was modified to remove some packages and the `.dockerignore` added the `examples` directory which was 300+MB. With the docker optimization, the original image went from 1.86GB to 1.33GB. Further reduction can be attained by not including the `.git` folder but the testing needs that for coveralls.

All tests pass in SMRF and the documentation builds in the docker image.

This has been tested to be compatible with the [IPW pull request](https://github.com/USDA-ARS-NWRC/ipw/pull/4)

### CODE REVIEW

**Comment by scotthavens** [comment]
This PR will also fix the security vulnerability and allow a different package to manage the version of `urllib3`

**Comment by micahjohnson150** [comment]
Do you have thoughts on which version of setup tools you want to use? I ask because the newest version is deprecating python setup.py test and is asking folks to switch to tox

**Comment by scotthavens** [comment]
Then probably the latest version that supports `python setup.py test` or we switch to `python -m unittest` which is a little more verbose since we are not testing different types of python versions.

**Comment by micahjohnson150** [comment]
This looked good to me, who else do you want to review?

**Comment by micahjohnson150** [comment]
@scotthavens should we merge this in? Or should we wait for @etrujil  to review? I just updated this with the new changes from PR #133

**Comment by etrujil** [comment]
I would happily review but would need some background as for what kind of stuff I should be looking at. Just let me know. @scotthavens @micahjohnson150

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ services:
   - docker
 
 before_install:
-  - docker build -t usdaarsnwrc/smrf .
+  - docker build --build-arg REQUIREMENTS=_dev -t usdaarsnwrc/smrf .
```
Comment (etrujil) [inline .travis.yml]: What does this do?

**Pull request merged**